### PR TITLE
Floor types

### PR DESCRIPTION
**Pull request by bpark1327**
## Pull Request Description

Allows for describing additional floor types including SIP, concrete, steel-frame floors.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [x] Schematron validator (`EPvalidator.xml`) has been updated
- [x] Sample files have been added/updated (via `tasks.rb`)
- [x] Unit tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests`)
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] Checked the code coverage report on CI
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: BuildResidentialHPXML/measure.xml [hunk 1]
```diff
@@ -6440,7 +6440,11 @@
       <filename>measure.rb</filename>
       <filetype>rb</filetype>
       <usage_type>script</usage_type>
+<<<<<<< HEAD
+      <checksum>17D78430</checksum>
+=======
       <checksum>959A3C05</checksum>
+>>>>>>> 45aacfa870089762f2f170a671a8a6186c0da641
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.xml:6447]: @bpark1327 Looks like some conflicts weren't resolved correctly here.

**Comment by shorowit** [comment]
@bpark1327 Need to update the Changelog to call out the new feature/breaking change.

**Comment by shorowit** [comment]
@bpark1327 Can you pull in the latest `master` code?

File: HPXMLtoOpenStudio/resources/constructions.rb [hunk 2]
```diff
@@ -1662,6 +1889,139 @@ def self.apply_wall_construction(runner, model, surfaces, wall_id, wall_type, as
...
+          SteelStudConstructionSet.new(5.5, corr_factor, 0.10, 40.0, 0.0, mat_int_finish, nil),         # 2x6, 24" o.c. + R40
+          SteelStudConstructionSet.new(5.5, corr_factor, 0.10, 30.0, 0.0, mat_int_finish, nil),         # 2x6, 24" o.c. + R30
+          SteelStudConstructionSet.new(5.5, corr_factor, 0.10, 20.0, 0.0, mat_int_finish, nil),         # 2x6, 24" o.c. + R20
+          SteelStudConstructionSet.new(5.5, corr_factor, 0.10, 10.0, 0.0, mat_int_finish, nil),         # 2x6, 24" o.c. + R10
+          SteelStudConstructionSet.new(3.5, corr_factor, 0.13, 0.0, 0.0, mat_int_finish, nil),          # 2x4, 16" o.c.
+          SteelStudConstructionSet.new(3.5, 1.0, 0.01, 0.0, 0.0, fallback_mat_int_finish, nil), # Fallback
+        ]
+        match, constr_set, cavity_r = pick_steel_stud_construction_set(assembly_r, constr_sets, inside_film, outside_film)
+
+        apply_steel_stud_ceiling(model, surface, "#{floor_id} construction",
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/constructions.rb:1956]: steel_stud -> steel_frame

**Review by shorowit — COMMENTED** [review]
Looks pretty good. It's great to see no diffs aside from the new test HPXML. Here are a few initial review comments.

**Review by shorowit — COMMENTED** [review]
This also needs some tests. You can add floor/ceiling tests into [here](https://github.com/NREL/OpenStudio-HPXML/blob/515cdc37746d219684a23192a5f50962bbfdafd9/HPXMLtoOpenStudio/tests/test_enclosure.rb#L360) and follow the pattern used for walls [here](https://github.com/NREL/OpenStudio-HPXML/blob/515cdc37746d219684a23192a5f50962bbfdafd9/HPXMLtoOpenStudio/tests/test_enclosure.rb#L230-L289).

File: HPXMLtoOpenStudio/resources/constructions.rb [hunk 3]
```diff
@@ -1662,6 +1889,139 @@ def self.apply_wall_construction(runner, model, surfaces, wall_id, wall_type, as
...
+        constr_sets = [
+          WoodStudConstructionSet.new(Material.Stud2x6, 0.10, 20.0, 0.75, nil, mat_ext_finish),        # 2x6, 24" o.c. + R20
+          WoodStudConstructionSet.new(Material.Stud2x6, 0.10, 10.0, 0.75, nil, mat_ext_finish),        # 2x6, 24" o.c. + R10
+          WoodStudConstructionSet.new(Material.Stud2x6, 0.10, 0.0, 0.75, nil, mat_ext_finish),         # 2x6, 24" o.c.
+          WoodStudConstructionSet.new(Material.Stud2x4, 0.13, 0.0, 0.5, nil, mat_ext_finish),          # 2x4, 16" o.c.
+          WoodStudConstructionSet.new(Material.Stud2x4, 0.01, 0.0, 0.0, nil, fallback_mat_ext_finish), # Fallback
+        ]
+        match, constr_set, cavity_r = pick_wood_stud_construction_set(assembly_r, constr_sets, inside_film, outside_film)
+
+        apply_wood_stud_floor(model, surface, "#{floor_id} construction",
```
[the remark this answers is not in the document]
Comment (bpark1327) [reply HPXMLtoOpenStudio/resources/constructions.rb:1934]: Okay. I will separate the method for the floor and ceiling.

**Comment by bpark1327** [comment]
@shorowit I can't understand why my last commit failed. Please advise.

**Comment by shorowit** [comment]
> @shorowit I can't understand why my last commit failed. Please advise.

You can safely ignore the windows CI error, the OpenStudio team removed the rc2 build it was trying to use. The next time you pull in master (which switches to the official OS 3.5 release), the error will be resolved.

**Review by shorowit — APPROVED** [review]
Looks good to me. Thanks, @bpark1327!

**Pull request merged**